pydantic/types.py

```diff
@@ -1055,3 +1056,74 @@ def validate(cls, value: date) -> date:
                 raise errors.DateNotInTheFutureError()
 
             return value
+
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ NANO TIME TYPE ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+TIME_UNITS = {
+    'ns': 1,
+    'ms': 10 ** 6,
+    's': 10 ** 9,
+    'm': 60 * 10 ** 9,
+    'h': 60 * 60 * 10 ** 9,
+    'd': 24 * 60 * 60 * 10 ** 9
```

Comment:
what about "seconds", "secs", "mins", "minutes", "hours", "hrs", "days", even "microseconds"?

Could we add something like
```py
UNIT_LOOKUP = {
    'min': 'm',
    'minute': 'm',
    ...
}

...
            unit_l = unit.lower()
            try:
                unit_mult = TIME_UNITS[unit_l]
            except KeyError:
                try:
                    unit_mult = TIME_UNITS[UNIT_LOOKUP[unit_l.rstrip('s')]
                except KeyError:
                    raise errors.InvalidNanoTimeUnit(unit=unit)
```

or something?